code/libs/Get Flat Icon/Get Flat Icon.dm

```diff
@@ -133,7 +133,8 @@ var/list/exception = list(
 	for(var/I in layers)
 		if (!I:icon && !I:icon_state) // no icon nor icon_state? we're probably not meant to draw that. Possibly a blank icon while we're only interested in its overlays.
 			continue
-
+		if (I:blend_mode == BLEND_ADD)//additive overlays don't show up properly, especially if they're partly transparent
```

Comment:
using the unsafe trust operator isn't good (of course the author of GFI uses it all over the place because he's not obeying our standards); is there any justifiable reason not to simply cast as type?

Reply:
"I" here can be either an atom or an image. this removes the need for a typecheck that would further slow down an already pretty slow proc.

Reply:
So you can bypass typechecks with this?

Reply:
Technically yes but as Kurfurst said this doesn't follow our standards, so we avoid using it unless well justified.

Reply:
how does 'as' factor in?

Reply:
that's also something we've been phasing out from the args of proc definitions for the past decade. It's mostly found in oldcode now.

The only instance of "as" I know we keep using actively is when doing input(). As things like "as null|anything" let you cancel out of a choice, or "as color" let you use a color picker, etc.

Reply:
If there was a carbon/human but only referenced as a carbon in the var declaration, and you call a proc which is defined on carbon but overridden on carbon/human, which version of the proc gets called?

Reply:
the carbon/human version. It will use the version that matches the actual datum being passed as argument.